### ISSUE TITLE
[Bug]: Upload of multiple archives with the same filename fails

### ISSUE THREAD

**Issue by ngandrass** [ngandrass/moodle-quiz_archiver#55]
### Quiz archiver moodle plugin version

2.1.0

### Quiz archiver worker version

2.1.0

### Worker service deployment method

Docker compose

### Moodle Version

4.5

### PHP Version

8.3

### Database

PostgreSQL

### Operating system

Arch Linux

### What happened?

If a filename pattern for the archive file produces duplicate filenames, the first archive can be uploaded but subsequent archive jobs fail with the following error:

```text
quiz-archive-worker-1  | [2024-10-29 12:14:02,988] | ERROR    | archiveworker.quiz_archive_job::<7bebae2c-95e6-11ef-b857-0242c0a88002> | Job failed with error: RuntimeError: Moodle webservice failed to process uploaded artifact with status: E_STORE_ARTIFACT_FAILED
```

### Steps to reproduce

1. Create an archive job with the archive filename set to `test`
2. Create another archive job with the archive filename set to `test`

**Comment by ngandrass** [comment]
This bug was found by @melanietreitinger

**Comment by ngandrass** [comment]
Fixed with #56 and part of the next release :tada:
